iOS/Flows/Home/_New/new_ConversationViewController.swift

```diff
@@ -99,37 +87,68 @@ class new_ConversationViewController: FullScreenViewController, CollectionViewIn
             self.conversationController.deleteMessage(message.id)
         }
     }
-}
-
-extension new_ConversationViewController: UICollectionViewDelegate {
 
+    // MARK: - UICollectionViewDelegate
 
+    /// If true, the conversation controller is currently loading messages.
+    @Atomic private var isLoadingMessages = false
     func collectionView(_ collectionView: UICollectionView,
                         willDisplay cell: UICollectionViewCell,
                         forItemAt indexPath: IndexPath) {
 
-        // If all the messages are loaded, no need to fetch more.
+        // If all the messages are loaded, there's no need to fetch more.
         guard !self.conversationController.hasLoadedAllPreviousMessages else { return }
 
         // Start fetching new messages once the user is nearing the end of the list.
-        guard indexPath.row < 10 else { return }
+        guard indexPath.row < 2 else { return }
 
         Task {
+            guard !isLoadingMessages else { return }
+
+            self.isLoadingMessages = true
             do {
                 let oldestMessageID = self.conversationController.messages.first?.id
                 try await self.conversationController.loadPreviousMessages(before: oldestMessageID)
             } catch {
                 logDebug(error)
             }
+            self.isLoadingMessages = false
         }
     }
-}
 
-extension new_ConversationViewController {
+    func scrollViewWillEndDragging(_ scrollView: UIScrollView,
+                                   withVelocity velocity: CGPoint,
+                                   targetContentOffset: UnsafeMutablePointer<CGPoint>) {
 
-    func subscribeToConversationUpdates() {
-        self.conversationController?.delegate = self
+        // Always scroll so that a message is centered when we stop scrolling.
+        var newXOffset = CGFloat.greatestFiniteMagnitude
+        let targetOffset = targetContentOffset.pointee
+
+        let targetRect = CGRect(x: targetOffset.x,
+                                y: targetOffset.y,
+                                width: scrollView.width,
+                                height: scrollView.height)
+
+        let layout = self.collectionView.collectionViewLayout
+        guard let layoutAttributes = layout.layoutAttributesForElements(in: targetRect) else {
+            return
+        }
+
+        // Find the item whose center is closest to the proposed offset
+        for elementAttributes in layoutAttributes {
+            let possibleNewOffset = elementAttributes.frame.centerX - collectionView.halfWidth
+            if abs(possibleNewOffset - targetOffset.x) < abs(newXOffset - targetOffset.x) {
+                newXOffset = possibleNewOffset
+            }
+        }
+
+        targetContentOffset.pointee = CGPoint(x: newXOffset, y: targetOffset.y)
```

Comment:
Long story short, this ensures we always land on a centered cell (as much as is possible, content insets might make get in the way, but it'll at least be on screen).